verilog/analysis/verilog_project.cc

```diff
@@ -195,11 +197,22 @@ absl::StatusOr<VerilogSourceFile*> VerilogProject::OpenIncludedFile(
   return inserted.first->second->Status();
 }
 
-void VerilogProject::AddVirtualFile(absl::string_view referenced_filename,
+void VerilogProject::AddVirtualFile(absl::string_view resolved_filename,
```

Comment:
Here I have some questions that might be clarified with comments and unit tests.

The implementation makes the assumption that, given a resolved name, we can get a 'shorter' referenced name because it is considered that one of the includes has the prefix, if not we key it by full resolved name. Could this be a problem ? Maybe a comment about the reasoning would be good.

It looks like we now key all files by referenced name, i.e. basename. Couldn't this result in issues if we have multiple files with the same basename but different paths ?

Let's say there are files
```
/foo/bar/baz.sv
/abc/xyz/baz.sv
```

and include paths `.`, `/foo/bar` and `/abc/xyz`. This `AddVirtualFile()` seems to add these under the key `bar.sv` (and possibly overwriting two `bar.sv` with last one winning ?).

Depending on the context, `baz.sv` might also need to be resolved differently. Say we have a file `/foo/bar/hello.sv` and `/abc/xyz/world.sv` which each include `baz.sv`.

Given the include search path (that has as first element `.`), the first one should open `/foo/bar/baz.sv`, the second one `/abc/xyz/baz.sv`. Will this work ?

I suspect that instead of the reference filename we need to key the files with the full filename ('resolved_filename') and whenever we get a reference file to open resolve it with the current context, which includes the dirname of the file we resolve it from (in which `.` has a different meaning).

Reply:
I agree that would be a better approach. But the includes are opened while processing AST. We don't know the current directory of the file that lists the include. If `.` is an implicit include dir as in C++, then we need to rework the AST visitor to propagate the current filename.

Let me change the PR back to resolving at the _opening_ time instead the _file adding_ time. The last time I missed to change `OpenTranslationUnit`. Let's see how the whole pipeline works if I change both of them.

Reply:
Pushed a commit to resolve includes at the opening time instead the file adding time. This doesn't include the current directory, as that requires more plumbing.